### ISSUE TITLE
React Hooks

### ISSUE THREAD

**Issue by engmsilva** [express-labs/pure-react-carousel#159]
First congratulations on the project example. I would like to know how do I listen to component events using a React Hooks structure. I know there is the HOC WithStore () component to listen for events, but I'm having a hard time implementing React Hooks. Would there be any simple examples for me to have a direction?

**Comment by mrbinky3000** [comment]
All the components in Pure React Carousel communicate with each other using
React's context API.  I believe you can use "useContext"  I haven't tried
it yet myself.

That being said, you CAN use higher order components to pass props to
purely presentational react components. You might not need to use hooks.
The new shinyt !== right tool for the job every time. :-)

**Comment by mrbinky3000** [comment]
Can you show me a sample of your code?

**Comment by engmsilva** [comment]
@mrbinky3000 Thanks for the feedback. As you can see in the code, I can't hear the currentSlide event, it returns undefined.

`const CarouselComponent = (props, {state}) => {  
  const { data, idFoto, goToPrevSlide, goToNextSlide } = useCustomHooks(props);
  const { currentSlide } = state();
  console.log(currentSlide) // return undefined

  return (
    <div>
      <CarouselProvider
        visibleSlides={1}
        totalSlides={data.length}
        step={1}
        dragStep={2}
        naturalSlideWidth={100}
        naturalSlideHeight={100}
        hasMasterSpinner
      >
        <div className="container">
          <Slider className="slider">
            {data.length
      && data.map((image, i) => (
        <Slide index={i}>

          <ImageWithZoom src={image.foto} style={{ filter: 'grayscale(100%)' }} />
          <div className="titleSlide">{image.animal}</div>
        </Slide>
      ))}

          </Slider>         
          <ButtonBack className="buttonBack" onClick={() => goToPrevSlide()}><KeyboardArrowLeft /></ButtonBack>
          <ButtonNext className="buttonNext" onClick={() => goToNextSlide()}> <KeyboardArrowRight /></ButtonNext>

        </div>
      </CarouselProvider>
      <Wrapper idFoto={idFoto} enable={data.length > 0} />
    </div>

  )}

  export default WithStore(CarouselComponent, state => ({ 
  currentSlide: state.currentSlide,
  )}`

**Comment by mrbinky3000** [comment]
A couple of things:

The following HOC...
```
export default WithStore(CarouselComponent, state => ({
  currentSlide: state.currentSlide,
)}
```
Will read the current context for the current slide, and then pass a prop called currentSlide to your CarouselComponent functional component. It's just a simple prop at this point.  No context.  Definitely not state.  Think of the `WithStore()` HOC as a bridge between Pure React Carousel's context and props.

You seem to be looking for currentSlide in state.
```
const { currentSlide } = state();
```
It's not state, it's just a regular prop.

**Comment by mrbinky3000** [comment]
I edited your post so I can see the javascript better.  In doing so, I could see all the code.

I now know of another problem with your code.  You're trying to use the WithStore() HOC in the wrong location.   Any use of WithStore must occur in a component that is a descendant of `<CarouselProvider />`

The Carousel Provider sets the context used by Pure React Carousel.  React context is only made available to child (descendant) components.  As your code is written `{ current Slide }` is undefined because, outside of Carousel Provider, it IS undefined.

Closing for now.  Reopen if you have any questions.
